File: May-18/09-05-18/python/01-Shubham-and-Xor.py
n = int(input())
ar = [int(x) for x in input().split()]
ar.sort()
tmp = -1
cnt = 1
ans = 0

for i in range(0, n):
    if(tmp != ar[i]):
        tmp = ar[i]
        ans += (cnt*(cnt-1))//2
        cnt = 1
    else:
        cnt += 1

ans += (cnt*(cnt-1))//2
print(ans)

File: May-18/09-05-18/c++/01-Shubham-and-Xor.cpp
#include<bits/stdc++.h>

using namespace std;
typedef long long ll;

int main()
{
    int n;
    cin >> n;
    int ar[n];
    ll ans = 0;
    for(int i = 0; i < n; i++)
    {
        cin >> ar[i];
    }
    sort(ar, ar+n);
    
    int tmp = -1;
    ll cnt = 1;
    for(int i = 0; i < n; i++)
    {
        if(ar[i] != tmp)
        {
            tmp = ar[i];
            ans += (cnt*(cnt-1))/2;
            cnt = 1;
        }
        else
        {
            cnt += 1;
        }
        
    }
    ans += (cnt*(cnt-1))/2;
    
    cout << ans << endl;
}
